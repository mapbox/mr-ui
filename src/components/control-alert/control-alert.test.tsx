import React from 'react';
import ControlAlert from './control-alert';
import { screen, render, fireEvent } from '@testing-library/react';

describe('ControlAlert', () => {
  describe('basic error', () => {
    const props = {
      children: <span className="ml6">Drats.</span>,
      theme: 'error'
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlAlert {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('basic locked', () => {
    const props = {
      children: <span className="ml6">Nope.</span>,
      theme: 'locked'
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlAlert {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('basic success', () => {
    const props = {
      children: <span className="ml6">Woohoo!</span>,
      theme: 'success'
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlAlert {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('dismissive warning', () => {
    const mockOnClick = jest.fn();

    const props = {
      children: <span className="mx6">Uh oh...</span>,
      onButtonClick: mockOnClick,
      theme: 'warning'
    } as const;

    test('onButtonClick is called', () => {
      render(<ControlAlert {...props} />)
      fireEvent.click(screen.getByTestId('alert-dismiss'));
      expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
  });

  describe('dismissive multiline error', () => {
    const mockOnClick = jest.fn();

    const props = {
      children: (
        <p className="mx6">
          This is the song that never ends.{' '}
          <a className="link" href="http://www.mapbox.com">
            Yes, it just goes on and on my friends.
          </a>{' '}
          Some people started singing it, not knowing what it was. And they'll
          continue singing it forever just because. This is the song that never
          ends.
        </p>
      ),
      onButtonClick: mockOnClick,
      theme: 'error'
    } as const;

    test('onButtonClick is called', () => {
      render(<ControlAlert {...props} />)
      fireEvent.click(screen.getByTestId('alert-dismiss'));
      expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
  });
});
