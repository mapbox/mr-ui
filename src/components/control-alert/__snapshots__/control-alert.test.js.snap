// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlAlert basic error renders as expected 1`] = `
<div
  className="round px24 py24 flex bg-red-faint"
>
  <div
    className="flex-child-no-shrink"
  >
    <Icon
      inline={true}
      name="alert"
      size="1.3em"
    />
  </div>
  <div
    className="flex-child-grow"
    role={null}
    tabIndex={null}
  >
    <span
      className="ml6"
    >
      Drats.
    </span>
  </div>
</div>
`;

exports[`ControlAlert basic locked renders as expected 1`] = `
<div
  className="round px24 py24 flex bg-yellow-light"
>
  <div
    className="flex-child-no-shrink"
  >
    <Icon
      inline={true}
      name="lock"
      size="1.3em"
    />
  </div>
  <div
    className="flex-child-grow"
    role={null}
    tabIndex={null}
  >
    <span
      className="ml6"
    >
      Nope.
    </span>
  </div>
</div>
`;

exports[`ControlAlert basic success renders as expected 1`] = `
<div
  className="round px24 py24 flex bg-green-light"
>
  <div
    className="flex-child-no-shrink"
  >
    <Icon
      inline={true}
      name="check"
      size="1.3em"
    />
  </div>
  <div
    className="flex-child-grow"
    role={null}
    tabIndex={null}
  >
    <span
      className="ml6"
    >
      Woohoo!
    </span>
  </div>
</div>
`;

exports[`ControlAlert dismissive multiline error onButtonClick is called 1`] = `
<div
  className="round px24 py24 flex bg-red-faint"
>
  <div
    className="flex-child-no-shrink"
  >
    <Icon
      inline={true}
      name="alert"
      size="1.3em"
    />
  </div>
  <div
    className="flex-child-grow"
    role={null}
    tabIndex={null}
  >
    <p
      className="mx6"
    >
      This is the song that never ends.
       
      <a
        className="link"
        href="http://www.mapbox.com"
      >
        Yes, it just goes on and on my friends.
      </a>
       
      Some people started singing it, not knowing what it was. And they'll continue singing it forever just because. This is the song that never ends.
    </p>
  </div>
  <Tooltip
    block={true}
    content="Dismiss"
  >
    <button
      aria-label="Dismiss"
      className="bg-transparent color-gray-dark color-blue-on-hover px0 py0"
      data-test="alert-dismiss"
      onClick={[Function]}
    >
      <Icon
        inline={true}
        name="close"
        size="1.3em"
      />
    </button>
  </Tooltip>
</div>
`;

exports[`ControlAlert dismissive multiline error renders as expected 1`] = `
<div
  className="round px24 py24 flex bg-red-faint"
>
  <div
    className="flex-child-no-shrink"
  >
    <Icon
      inline={true}
      name="alert"
      size="1.3em"
    />
  </div>
  <div
    className="flex-child-grow"
    role={null}
    tabIndex={null}
  >
    <p
      className="mx6"
    >
      This is the song that never ends.
       
      <a
        className="link"
        href="http://www.mapbox.com"
      >
        Yes, it just goes on and on my friends.
      </a>
       
      Some people started singing it, not knowing what it was. And they'll continue singing it forever just because. This is the song that never ends.
    </p>
  </div>
  <Tooltip
    block={true}
    content="Dismiss"
  >
    <button
      aria-label="Dismiss"
      className="bg-transparent color-gray-dark color-blue-on-hover px0 py0"
      data-test="alert-dismiss"
      onClick={[Function]}
    >
      <Icon
        inline={true}
        name="close"
        size="1.3em"
      />
    </button>
  </Tooltip>
</div>
`;

exports[`ControlAlert dismissive warning onButtonClick is called 1`] = `
<div
  className="round px24 py24 flex bg-yellow-light"
>
  <div
    className="flex-child-no-shrink"
  >
    <Icon
      inline={true}
      name="alert"
      size="1.3em"
    />
  </div>
  <div
    className="flex-child-grow"
    role={null}
    tabIndex={null}
  >
    <span
      className="mx6"
    >
      Uh oh...
    </span>
  </div>
  <Tooltip
    block={true}
    content="Dismiss"
  >
    <button
      aria-label="Dismiss"
      className="bg-transparent color-gray-dark color-blue-on-hover px0 py0"
      data-test="alert-dismiss"
      onClick={[Function]}
    >
      <Icon
        inline={true}
        name="close"
        size="1.3em"
      />
    </button>
  </Tooltip>
</div>
`;

exports[`ControlAlert dismissive warning renders as expected 1`] = `
<div
  className="round px24 py24 flex bg-yellow-light"
>
  <div
    className="flex-child-no-shrink"
  >
    <Icon
      inline={true}
      name="alert"
      size="1.3em"
    />
  </div>
  <div
    className="flex-child-grow"
    role={null}
    tabIndex={null}
  >
    <span
      className="mx6"
    >
      Uh oh...
    </span>
  </div>
  <Tooltip
    block={true}
    content="Dismiss"
  >
    <button
      aria-label="Dismiss"
      className="bg-transparent color-gray-dark color-blue-on-hover px0 py0"
      data-test="alert-dismiss"
      onClick={[Function]}
    >
      <Icon
        inline={true}
        name="close"
        size="1.3em"
      />
    </button>
  </Tooltip>
</div>
`;
